ui/src/components/layouts/TheSignIn.vue

```diff
@@ -180,56 +182,89 @@ export default {
 	},
 	data() {
 		return {
-			data1: [],
-		}
+			data1: {},
+			msg: "",
+		};
+	},
+	created() {
+
 	},
 	methods: {
-		// tạm thời chưa viết xong phải quay lại viết chức năng này
+		/**
+		 * sign in click
+		 * CreatedBy: DucLM (20/09/2021)
+		 * 1. check null user password : not null then call api
+		 * 2. call api get user by username
+		 * 	- data is null then show message: "Tài khoản không tồn tại"
+		 * 3. check password
+		 * 	- true: go to HomePage
+		 *  - false: show msg
+		 */
 		btnSignInClick() {
+			// 1. check null user password
 			if (this.validate()) {
-				axios.get(`http://localhost:3000/users?username=${this.username}`)
+				//2 call api
+				axios
+					.get(`http://localhost:3000/users?username=${this.username}`)
 					.then((response) => {
```

Comment:
Xử lý response này em nên xử lý theo status_code. EM tìm hiểu luôn các status code thường gặp và ý nghĩa từng status nhé. Backend mình sẽ trả về theo satus code này và FE sẽ xử lý response dựa theo status code

Reply:
cái này em thử status_code rồi nhưng cái json-server nó toàn trả về status_code =200 mặc dù cái data rỗng nên em phải check tay như trong code